packages/office-ui-fabric-react/src/components/Dialog/Dialog.base.tsx

```diff
@@ -82,12 +83,21 @@ export class DialogBase extends BaseComponent<IDialogProps, {}> {
       topButtonsProps,
       type,
       minWidth,
-      maxWidth
+      maxWidth,
+      modalProps
     } = this.props;
 
-    const modalProps = {
+    const mergedLayerProps: ILayerProps = {
+      ...(modalProps ? modalProps.layerProps : { onLayerDidMount })
+    };
+    if (onLayerDidMount && !mergedLayerProps.onLayerDidMount) {
+      mergedLayerProps.onLayerDidMount = onLayerDidMount;
+    }
```

Comment:
Not sure... but I believe this logic is not accounting for `modalProps.onLayerDidMount` case. What about this old style approach with some nested if else statements. To me it looks more readable and it takes into account all cases, unless I am missing something. @natalieethell could you take a look 
```ts
let layerPropsFiltered: ILayerProps;
if (modalProps) {
    if (modalProps.layerProps) {
        layerPropsFiltered = modalProps.layerProps;
    } else {
        layerPropsFiltered = { onLayerDidMount: modalProps.onLayerDidMount };
    }
} else {
    layerPropsFiltered = { onLayerDidMount: onLayerDidMount };
}

const mergedModalProps = {
    ...DefaultModalProps,
    ...modalProps,
    layerProps: { ... layerPropsFiltered }
}
```

Reply:
You are right... didn't take that case into account. Thanks for getting a clear analysis of my comment. Your solution works now that I see where is the missing case I didn't notice 😄